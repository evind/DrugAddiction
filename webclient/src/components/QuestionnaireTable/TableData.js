import React from "react";

const tdStyle = {
  textAlign: "center",
};

class TableData extends React.Component {
  constructor(props) {
    super(props);

    this.state = { text: "" };
  }

  componentDidMount() {
    const name = this.props.name;
    const answer = this.props.ans;

    if (name == "Never" && answer == 1) {
      this.setState({ text: "X" });
    } else if (name == "Rarely" && answer == 2) {
      this.setState({ text: "X" });
    } else if (name == "Sometimes" && answer == 3) {
      this.setState({ text: "X" });
    } else if (name == "Fairly Often" && answer == 4) {
      this.setState({ text: "X" });
    } else if (name == "Often" && answer == 5) {
      this.setState({ text: "X" });
    } else if (name == "Almost Always" && answer == 6) {
      this.setState({ text: "X" });
    } else if (name == "Always" && answer == 7) {
      this.setState({ text: "X" });
    }
  }

  renderContent() {}

  render() {
    return (
      <td style={tdStyle} className="tData">
        {this.state.text}
      </td>
    );
  }
}

export default TableData;
