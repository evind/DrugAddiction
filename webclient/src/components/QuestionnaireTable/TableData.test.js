import TableData from "./TableData2";
import { render, fireEvent, screen } from "@testing-library/react";

describe("TableData", () => {
  it("should render", () => {
    const { asFragment } = render(<TableData name="Never" ans={1} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
