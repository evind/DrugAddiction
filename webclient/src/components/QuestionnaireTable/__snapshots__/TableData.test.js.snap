// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableData should render 1`] = `
<DocumentFragment>
  X
</DocumentFragment>
`;
